PhotoAndCopyShop.ScanWorker/FolderScannerDaemon.cs

```diff
@@ -0,0 +1,40 @@
+﻿using PhotoAndCopyShop.ScanWorker.Handlers;
+
+namespace PhotoAndCopyShop.ScanWorker;
+
+public class FolderScannerDaemon
+{
+    private readonly string _folderPath;
+    private readonly IHandler<FileInfo> _handler;
+    private readonly List<FileInfo> _scannedFiles = new();
+
+    public FolderScannerDaemon(string folderPath, IHandler<FileInfo> handler)
+    {
+        _folderPath = folderPath;
+        _handler = handler;
+    }
+
+    public async Task ScanAsync()
+    {
+        Console.WriteLine($"Scanning folder {_folderPath} for new files...");
+
+        while (true)
```

Comment:
Can you implement some kind of circuit breaker to stop this process in a safe manner?